Mathlib/RingTheory/Binomial.lean

```diff
@@ -176,6 +176,98 @@ instance Int.instBinomialRing : BinomialRing ℤ where
 
 end Nat_Int
 
+section Neg
+
+namespace Ring
+
+open Polynomial
+
+variable {R : Type*} [NonAssocRing R] [Pow R ℕ] [BinomialRing R]
+
+@[simp]
+theorem ascPochhammer_smeval_neg_self : ∀(n : ℕ),
+    smeval (ascPochhammer ℕ n) (-n : ℤ) = (-1)^n * n.factorial
+  | 0 => by
+    rw [Nat.cast_zero, neg_zero, ascPochhammer_zero, Nat.factorial_zero, smeval_one, pow_zero,
+      one_smul, pow_zero, Nat.cast_one, one_mul]
+  | n + 1 => by
+    rw [ascPochhammer_succ_left, smeval_X_mul, smeval_comp, smeval_add, smeval_X, smeval_one,
+      pow_zero, pow_one, one_smul, Nat.cast_add, Nat.cast_one, neg_add_rev, neg_add_cancel_comm,
+      ascPochhammer_smeval_neg_self n, ← mul_assoc, mul_comm _ ((-1) ^ n),
+      show (-1 + -↑n = (-1 : ℤ) * (n + 1)) by omega, ← mul_assoc, pow_add, pow_one,
+      Nat.factorial, Nat.cast_mul, ← mul_assoc, Nat.cast_succ]
+
+@[simp]
+theorem ascPochhammer_succ_smeval_neg (n : ℕ) :
+    smeval (ascPochhammer ℕ (n + 1)) (-n : ℤ) = 0 := by
+  rw [ascPochhammer_succ_right, smeval_mul, smeval_add, smeval_X, ← C_eq_natCast, smeval_C,
+    pow_zero, pow_one, Nat.cast_id, nsmul_eq_mul, mul_one, add_left_neg, mul_zero]
+
+theorem ascPochhammer_smeval_neg_add (n : ℕ) : ∀(k : ℕ),
+    smeval (ascPochhammer ℕ (n + k + 1)) (-n : ℤ) = 0
```

Comment:
All of these should be `smeval_ascPochhammer` rather than `ascPochhammer_smeval`.